### ISSUE TITLE
Feature Request : Support OS specific key bindings in single file

### ISSUE THREAD

**Issue by cideM** [shanalikhan/code-settings-sync#515]
The only issue I found where this functionality was mentioned in any way was #194 (all the way at the bottom).

I use a Mac at work but Linux at home. Right now every keyboard shortcut I add needs to be manually replicated across `keybindings.json` and `keybindingsMac.json`. I don't want shortcuts to be separated though. Instead, **I want to have the option of storing all keyboard shortcuts in one file and have that synced across all my VSC instances**, regardless of OS. I'd be happy to help with a PR (I am assuming there is no rush as there does not seem to be a great demand). If there are other solutions to my issue, please let me know. I really tried finding some info but maybe I am just not very creative ;)

I would imagine something like `sync.universalKeybindings: true` or `sync.separateKeybindings: false` 🤷‍♂️

**Comment by shanalikhan** [comment]
Why do you want to keep Mac `keybindings` and Windows keybindings in same file?
What the scenario. ?
Would it work if you use Mac key commands in windows keybindings instead ? Wont code mark them as error.

**Comment by shanalikhan** [comment]
Closing this issue.
Let me know i will reopen it.

**Comment by cideM** [comment]
Sorry I went on holidays and forgot about this. The reason for this is that if I e.g., download a plugin and define a keyboard shortcut for it, I want this shortcut to be synced to both my Linux machine at home and my MBP at work without having to copy & paste the shortcut from `keybindingsMac.json` to `keybindings.json`.

Right now I need to manually keep both keybindings files in sync.

**Comment by shanalikhan** [comment]
It would be great if u can help me produce the scenario for you
Which extension and shortcuts which will behave same for mac and windows ?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cideM** [comment]
For example, I have this keybinding on Linux

```
    {
        "key": "ctrl+k tab",
        "command": "workbench.action.openRecent",
    },
```

and this on mac

```
    {
        "key": "cmd+.",
        "command": "findJump.activate",
        "when": "editorTextFocus"
    },
```

Right now I copy & paste the keybinding from one file to the other when I create it. I could also just have both in the same file (more convenient). Maybe copy & pasting is really an okay option here.

**Comment by AndreasBackx** [comment]
Would also want this behaviour. There also are keybindings that don't use OS specific keys that would benefit from this. Using this functionality only confuses the users because it adds something to the settings that VS Code doesn't do.

**Comment by cbliard** [comment]
This scenario interests me as well, I have common keybindings that I want to share between both OS (alt+q to reflow paragraph).

Also since version 1.27, [vscode supports platform specific bindings](https://code.visualstudio.com/updates/v1_27#_platform-specific-keybindings) with `isLinux`, `isMac` and `isWindows` in the `when` clause:

```
{
  "key": "ctrl+o",
  "command": "workbench.action.files.openFolder",
  "when": "!isMac"
},
{
  "key": "cmd+o",
  "command": "workbench.action.files.openFolder",
  "when": "isMac"
}
```

**Comment by howar31** [comment]
Agree with @cbliard 

Since v1.27, VSCode support OS specific key bindings now.  And it'll be great to keep all the bindings in one file and sync them together.

**Comment by connorshea** [comment]
>I would recommend it to keep this config a part of Setting Sync Global Setting instead of env. setting.

@shanalikhan are you saying you want a setting like `sync.universalKeybindings: true` or no?

**Comment by shanalikhan** [comment]
No, its the Gist Settings. 
I'm talking about adding a key in Settings Sync Global Settings
Read here:

https://github.com/shanalikhan/code-settings-sync#settings

**Comment by amccaugh** [comment]
Wanted to throw my support in for this feature request -- I only have a few keybindings, and none of them are OS-specific, so it's annoying to have to copy and paste the keybindings file back and forth.

**Comment by shanalikhan** [comment]
>Wanted to throw my support in for this feature request

@amccaugh 
Yes go ahead, PR is welcome. You need to create new configuration in Settings Sync Global Settings to use this method.
Will release new version

**Comment by njkevlani** [comment]
Hey guys, I have opened a PR #854 addressing this issue and some other features. We need some testers on **macOS**.

# Features to be tested is
 A new setting `sync.universalKeybindings` is introduced in [Global Settings ](https://github.com/shanalikhan/code-settings-sync#settings)of `Sync`

[Here](https://github.com/shanalikhan/code-settings-sync/pull/854#issuecomment-485232451) is the expected behavior.



# How to install extension with that patch
You can either
1. Download and install this package
OR
2. Fetch Code from my repository and compile the package

## 1. Download and install this package
Download the compiled package from [here](https://drive.google.com/open?id=1kEBg-Yne0SDI-NU5B9uFOK_z7P8ITGMM).

And Install package using instructions given at https://stackoverflow.com/questions/37071388/how-to-install-vscode-extensions-offline
## 2. Fetch Code from my repository and compile the package
### To Fetch Code:
`git clone https://github.com/njkevlani/code-settings-sync && git checkout pragmaUtil-kb`

### To Compile the code:
```
npm install -g vsce
cd path/to/code-settings-sync/
vsce package
```

### To install the compiled package
https://stackoverflow.com/questions/37071388/how-to-install-vscode-extensions-offline

**Comment by shanalikhan** [comment]
Thanks @njkevlani 

I will release new version once the [build](https://drive.google.com/open?id=1kEBg-Yne0SDI-NU5B9uFOK_z7P8ITGMM) by @njkevlani  works for Mac users.

**Comment by shanalikhan** [comment]
I have released new version of Settings Sync v3.3.0 fixing this issue. Im closing this ticket let me know if u guys run into some problem
